File: BMICalculator/app/src/main/java/com/s15851/bmi_calc/BmrActivity.java
package com.s15851.bmi_calc;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

public class BmrActivity extends AppCompatActivity {

    Button launchDiet, btCalculateBmr;
    TextView bmrResultTextView;
    EditText weightTextView;
    EditText heightTextView;
    EditText ageTextView;



    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_bmr);

        launchDiet = findViewById(R.id.launchDiet);
        btCalculateBmr = findViewById(R.id.calcBmr);
        bmrResultTextView = findViewById(R.id.bmrResult);
        weightTextView = findViewById(R.id.weight);
        heightTextView = findViewById(R.id.height);
        ageTextView = findViewById(R.id.age);

        btCalculateBmr.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                showBmr();
            }
        });

        Button back = findViewById(R.id.btBack);
        back.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(getApplicationContext(), BmiActivity.class);
                startActivity(intent);
            }
        });

    }

    private void showBmr()
    {
        String inputWeight = weightTextView.getText().toString();
        String inputHeight = heightTextView.getText().toString();
        String inputAge = ageTextView.getText().toString();

        if (!inputHeight.isEmpty() && !inputWeight.isEmpty() && !inputAge.isEmpty()) {
            float heightValue = Float.parseFloat(inputHeight);
            float weightValue = Float.parseFloat(inputWeight);
            int ageValue = Integer.parseInt(inputAge);

            calculateBmr(weightValue, heightValue, ageValue);
        }
    }
    public void calculateBmr(double weight, double height, double age)
    {
        double bmrResult;

        bmrResult = 66.4730 + 13.7516 * weight + 5.033 * height - 6.7550 * age;

        bmrResultTextView.setText(String.valueOf(Math.round(bmrResult))+" calories");
    }
}

File: BMICalculator_Kotlin/app/src/main/java/com/s15851/bmicalculator/BmrActivity.kt
package com.s15851.bmicalculator

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_bmr.*
import kotlin.math.roundToInt

class BmrActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_bmr)

        calcBmr.setOnClickListener { showBmr() }

        btBack.setOnClickListener {
            val intent = Intent(this, BmiActivity::class.java)
            startActivity(intent)
        }
    }

    private fun showBmr() {
        val inputWeight = weight.text.toString()
        val inputHeight = height.text.toString()
        val inputAge = age.text.toString()

        if (inputHeight.isNotEmpty() && inputWeight.isNotEmpty() && inputAge.isNotEmpty()) {
            val heightValue = inputHeight.toDouble()
            val weightValue = inputWeight.toDouble()
            val ageValue = inputAge.toInt()
            calculateBmr(weightValue, heightValue, ageValue)
        }
    }

    private fun calculateBmr(weight: Double, height: Double, age: Int) {
        val bmrValue: Double = 66.4730 + 13.7516 * weight + 5.033 * height - 6.7550 * age
        bmrResult.text = bmrValue.roundToInt().toString() + " calories"
    }
}